### PR TITLE
Document module packages

### PR DESCRIPTION
**Pull request by cfa**
These commits add a new language documentation page covering _Module Packages_.

RfC: @niner, @ugexe.

CC: @moritz, @zoffixznet.

### CODE REVIEW

File: doc/Language/module-packages.pod6 [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+
+I<N.B.> "Module" is an overloaded term in Perl 6; this document
+focuses on use of the C<module> declarator.
+
+
+=head1 What are modules?
+
+Modules, like classes and grammars, are a kind of
+L<package|/language/packages>. Fundamentally, they are metaclasses
```
Comment (niner) [inline doc/Language/module-packages.pod6]: Modules are not metaclasses. They are named containers for code and data. The module object itself is just an instance of the ModuleHOW metaclass.

**Comment by niner** [comment]
All in all it reads like a great explanation of what it is and how it works. To become perfect, it would have to answer the question of why. When would one declare a module instead of e.g. a class or instead of just putting code in a file? But that's just a hint on how to improve it further. Certainly nothing that should prevent these comments from getting merged.

File: doc/Language/module-packages.pod6 (hunk 1 above)
Comment (cfa) [reply doc/Language/module-packages.pod6]: Thanks; I've pushed 71cf557 to clarify this point. Better?

**Comment by cfa** [comment]
@niner please take a look at 71cf557 re: metaclasses.

**Comment by niner** [comment]
Sounds better, yes. Thanks!

**Comment by cfa** [comment]
@niner Grand, thanks for reviewing. Okay for me to merge?

**Comment by niner** [comment]
Sure :)

**Pull request merged**